internal/driver/driver.go

```diff
@@ -67,39 +68,49 @@ func (s *Driver) HandleReadCommands(deviceName string, protocols map[string]mode
 
 	res = make([]*dsModels.CommandValue, len(reqs))
 
+	var device_name string
+	var baud_rate int
+
+	for i, protocol := range protocols {
+		device_name = fmt.Sprintf("%v", protocol["deviceName"])
+		baud_rate, _ = cast.ToIntE(protocol["baudRate"])
+
+		s.lc.Infof(fmt.Sprintf("Driver.HandleReadCommands(): protocol = %v, device_name = %v, baud_rate = %v", i, device_name, baud_rate))
+	}
+
 	for i, req := range reqs {
 		s.lc.Infof(fmt.Sprintf("Driver.HandleReadCommands(): protocols: %v resource: %v attributes: %v", protocols, req.DeviceResourceName, req.Attributes))
 
 		key_type_value := fmt.Sprintf("%v", req.Attributes["type"])
 
 		if key_type_value == "generic" {
-			key_dev_value := fmt.Sprintf("%v", req.Attributes["dev"])
-
-			key_baud_value, _ := cast.ToIntE(req.Attributes["baud"])
 			key_maxbytes_value, _ := cast.ToIntE(req.Attributes["maxbytes"])
 			key_timeout_value, _ := cast.ToIntE(req.Attributes["timeout"])
 
 			// check device is already initialized
-			if _, ok := s.generic[key_dev_value]; ok {
-				s.lc.Infof(fmt.Sprintf("Driver.HandleReadCommands(): Device %v is already initialized with baud - %v, maxbytes - %v, timeout - %v", s.generic[key_dev_value], key_baud_value, key_maxbytes_value, key_timeout_value))
+			if _, ok := s.generic[device_name]; ok {
+				s.lc.Infof(fmt.Sprintf("Driver.HandleReadCommands(): Device %v is already initialized with baud - %v, maxbytes - %v, timeout - %v", s.generic[device_name], baud_rate, key_maxbytes_value, key_timeout_value))
 			} else {
 				// initialize device for the first time
-				s.generic[key_dev_value] = NewUartGeneric(key_dev_value, key_baud_value, key_timeout_value)
-				s.generic[key_dev_value].rxbuf = nil
+				s.generic[device_name] = NewUartGeneric(device_name, baud_rate, key_timeout_value)
+				s.generic[device_name].rxbuf = nil
 
-				s.lc.Infof(fmt.Sprintf("Driver.HandleReadCommands(): Device %v initialized for the first time with baud - %v, maxbytes - %v, timeout - %v", s.generic[key_dev_value], key_baud_value, key_maxbytes_value, key_timeout_value))
+				s.lc.Infof(fmt.Sprintf("Driver.HandleReadCommands(): Device %v initialized for the first time with baud - %v, maxbytes - %v, timeout - %v", s.generic[device_name], baud_rate, key_maxbytes_value, key_timeout_value))
 			}
 
-			if err := s.generic[key_dev_value].GenericUartRead(key_maxbytes_value); err != nil {
+			err := s.generic[device_name].GenericUartRead(key_maxbytes_value)
+			if err == io.EOF {
+				s.lc.Infof(fmt.Sprintf("Driver.HandleReadCommands(): EOF reached"))
```

Comment:
Shouldn't this be consider and error which should be returned. like before?

Reply:
Agree (for this feature), it has to be returned like before and reverted the change. It was added as EOF error is not considered as actual error when returned by Read(). This change may require along with read and write integer.